src/drivers/riscv_plic0.c

```diff
@@ -89,18 +91,25 @@ void __metal_driver_riscv_plic0_init (struct metal_interrupt *controller)
 
 	for(int parent = 0; parent < __METAL_PLIC_NUM_PARENTS; parent++) {
 	    num_interrupts = __metal_driver_sifive_plic0_num_interrupts(controller);
+            assert(num_interrupts < PLIC0_MAX_INTERRUPTS);
```

Comment:
We don't use assertions anywhere else (as far as I know), so I hesitate to start adding them unless we have a plan for how to deal with them when they fail.